### PR TITLE
Revert 5412 url reader special chars

### PR DESCRIPTION
**Pull request by ales-erjavec**
##### Issue

Fixes https://github.com/biolab/orange3/issues/5723

##### Description of changes

* Revert gh-5412
* Tolerantly but properly encode url from the user input.

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

File: Orange/widgets/data/owfile.py [hunk 1]
```diff
@@ -470,6 +470,10 @@ def _get_reader(self) -> FileFormat:
             return reader
         else:
             url = self.url_combo.currentText().strip()
+            # tolerant parse, also prevent parsing as local file
+            parsed = QUrl.fromUserInput(url, os.devnull)
+            if parsed.isValid():
+                url = bytes(parsed.toEncoded()).decode("ascii")
```
Comment (VesnaT) [inline Orange/widgets/data/owfile.py:476]: The problem of this solution is, that the UrlReader itself still doesn't work, so the change would have to be made in every widget that uses it.

I'd rather the fix was made within the UrlReader.

Comment (ales-erjavec) [reply Orange/widgets/data/owfile.py:476]: No. UrlReader should not accept invalid urls and should not do anything that could break valid urls.

Comment (markotoplak) [reply Orange/widgets/data/owfile.py:476]: Yes, but valid at which level? Shouldn't we support URLs as a browser would? If there is some non-destructive encoding/decoding happening in the background, why not put it into `UrlReader`?

Comment (ales-erjavec) [reply Orange/widgets/data/owfile.py:476]: Fine.

**Pull request closed without merging**